pkg/services/validator/aws.go

```diff
@@ -801,6 +787,26 @@ func readDirectAwsCredentials(c *components.AWSPluginConfig) error {
 
 func readAwsCredsHelper(c *components.AWSPluginConfig) error {
 	var err error
+	profile, validate, err := clouds.ReadAwsProfile()
```

Comment:
nit: this would be a bit more concise if you pass the AWSPluginConfig into ReadAwsProfile and assign the fields directly instead of remapping via:

```go
if profile.AccessKeyID != "" {
	c.AccessKeyID = profile.AccessKeyID
	c.SecretAccessKey = profile.SecretAccessKey
	c.SessionToken = profile.SessionToken
	return nil
}
```

Reply:
my intention was to keep all state changes in one file, but for sure, I can simplify this!

Reply:
Thanks. That's just not a pattern that we use in this repo. Plus you're reading into a struct regardless. It's just a matter of whether you then map struct A onto struct B.